File: backend/src/main/java/com/stadium/booking/entity/Booking.java
package com.stadium.booking.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("booking")
public class Booking extends BaseEntity {
    private String bookingNo;
    private Long userId;
    private String userName;
    private String userPhone;
    private Long venueId;
    private String venueName;
    private Long courtId;
    private String courtName;
    private LocalDate bookingDate;
    private LocalTime startTime;
    private LocalTime endTime;
    private Integer slotCount;
    private Integer bookingType;
    private Long groupId;
    private Integer status;
    private String cancelReason;
    private LocalDateTime cancelledAt;
    private Long cancelledBy;
    private Integer cancelledByType;
    private LocalDateTime checkedInAt;
    private Long checkedInBy;
    private Integer checkinMethod;
    private LocalDateTime noShowMarkedAt;
    private Long noShowMarkedBy;
    private String remark;
}

File: miniapp/src/types/booking.ts
export interface Booking {
  id?: number
  bookingNo: string
  userId?: number
  userName?: string
  userPhone?: string
  venueId: number
  venueName: string
  courtId: number
  courtName: string
  bookingDate: string
  startTime: string
  endTime: string
  slotCount?: number
  bookingType?: number
  status: BookingStatus
  statusText?: string
  cancelReason?: string
  cancelledAt?: string
  checkedInAt?: string
  remark?: string
  createdAt: string
  qrToken?: string
}

export type BookingStatus = 1 | 2 | 3 | 4

export interface CreateBookingParams {
  venueId: number
  courtId: number
  bookingDate: string
  startTime: string
  endTime: string
  remark?: string
  bookingType?: number
}
